docs/frontend-system/architecture/05-extension-overrides.md

```diff
@@ -10,38 +10,93 @@ description: Frontend extension overrides
+  extensions: [apertureLightTheme, apertureDarkTheme]
+});
+```
 
-Mention that in can still be a good idea to split your overrides out into separate packages in large projects. But it's up to you to decide how to group the extensions into extension overrides.
+We exported the overrides for the light and dark app theme extensions from a separate file in the code snippet above. To override the default app's light and dark theme extensions, we must ensure that the extension namespace and name match the ones in the default app themes extension definitions.
 
- -->
+2. Use the overrides in your Backstage App
 
-## Overriding Existing Extensions
+Now we are able to use the overrides in a Backstage app:
 
-<!--
+```tsx
+// packages/app/src/App.tsx
+import { createApp } from '@backstage/frontend-app-api';
+import apertureOverrides from '@backstage/plugin-aperture-overrides'
 
-To override an existing extension you need to provide an extension through an extension override that has the same ID as the existing extension. That is, all of kind, namespace and name must match the extension you want to override.
+const app = createApp({
+ features: [ apertureOverrides ],
+});
 
-This means that you typically need to provide an explicit namespace when overriding extensions from a plugin.
+export default app.createRoot().
+```
 
- - Example of how to create an extension override for an existing extension from some plugin
+If the plugin you want to change is internal to your company or you just want to replace one of the application's core extensions, you can decide to store the overrides code directly in the app package or extract them to a separate package.
 
- -->
+Note that it can still be a good idea to split your overrides out into separate packages in large projects. But it's up to you to decide how to group the extensions into extension overrides.
+
+## Overriding Plugin Extensions
+
+To override an extension that is provided by a plugin, you need to provide an new extension that has the same ID as the existing extension. That is, all kind, namespace, and name options must match the extension you want to replace. This means that you typically need to provide an explicit namespace when overriding extensions from a plugin.
+
+Imagine you have a plugin with the ID `'search'`, and the plugin provides a page extension that you want to fully override with your own custom component. To do so, you need to create your page extension with an explicit `namespace` option that matches that of the plugin that you want to override, in this case `'search'`. If the existing extension also has an explicit `name` you'd need to set the `name` of your override extension to the same value as well.
+
+```tsx
+// packages/app/src/search.ts
+const customSearchPage = createPageExtension({
+  namespace: 'search',
+  // Omitting name since it is the index plugin page
+  defaultPath: 'search',
+  loader: () => Promise.resolve(<div>My custom search page</div>),
```

Comment:
```suggestion
  loader: async () => <div>My custom search page</div>,
```

this is a matter of taste, feel free to ignore

Reply:
Replace with a `import(...)`